src/openrct2/ride/Ride.cpp

```diff
@@ -4400,29 +4400,26 @@ static Vehicle* vehicle_create_car(
         // loc_6DDCA4:
         vehicle->TrackSubposition = VEHICLE_TRACK_SUBPOSITION_0;
         int32_t direction = tileElement->GetDirection();
-        x += word_9A3AB4[direction].x;
-        y += word_9A3AB4[direction].y;
-        z = tileElement->GetBaseZ();
-        vehicle->TrackLocation = { x, y, z };
+        auto chosenLoc = carPosition + CoordsXYZ{ word_9A3AB4[direction], 0 };
+        vehicle->TrackLocation = chosenLoc;
         vehicle->current_station = tileElement->AsTrack()->GetStationIndex();
 
-        z += RideTypeDescriptors[ride->type].Heights.VehicleZOffset;
+        chosenLoc.z += RideTypeDescriptors[ride->type].Heights.VehicleZOffset;
 
         vehicle->track_type = tileElement->AsTrack()->GetTrackType() << 2;
         vehicle->track_progress = 0;
         vehicle->SetState(VEHICLE_STATUS_MOVING_TO_END_OF_STATION);
         vehicle->update_flags = 0;
 
-        CoordsXY chosenLoc;
         // loc_6DDD26:
         do
         {
             vehicle->sprite_direction = scenario_rand() & 0x1E;
-            chosenLoc.y = y + (scenario_rand() & 0xFF);
```

Comment:
This is not the same. Previously, it would take the value `x` and `y` had before the loop and then add a random number, but now they accumulate.

Reply:
Nice catch